### PR TITLE
helm: Remove default cpu limit inside chart

### PR DESCRIPTION
**Pull request by RomainBelorgey**
**What type of PR is this?**

* fix: allow to have no cpu limit when chart used as dependency

**What this PR does / why we need it**:

* Removing the default cpu limit as can't be removed if used as dependency
* CPU limit should be removed on every deployment (not requests)
  [#1](https://medium.com/omio-engineering/cpu-limits-and-aggressive-throttling-in-kubernetes-c5b20bd8a718)
  [#2](https://www.howtogeek.com/devops/should-you-set-kubernetes-cpu-limits/)
  [#3](https://home.robusta.dev/blog/stop-using-cpu-limits)
* Other examples / writings can be found around that, still possible for someone to add back the cpu limit

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Comment by nezdolik** [comment]
@arkodg currently is not possible to have cpu limit empty(unlimited), since value of `500m` gets merged from parent chart. This is our values.yaml:
```
      resources:
        limits:
          memory: 1024Mi
        requests:
          cpu: 500m
          memory: 256Mi
```
This is the resulting config of a running pod:
```
    Limits:
      cpu:     500m
      memory:  1Gi
    Requests:
      cpu:      500m
      memory:   256Mi
```

**Comment by arkodg** [comment]
hey @RomainBelorgey good to see you here :) 
can you run `make generate` && `make testdata` && `make manifests` and commit those changes ?

**Comment by RomainBelorgey** [comment]
> hey @RomainBelorgey good to see you here :) can you run `make generate` && `make testdata` && `make manifests` and commit those changes ?

Good to see you too :)

Thanks, I just run these commands and pushed the changes

**Review by arkodg — DISMISSED** [review]

File: test/helm/gateway-addons-helm/default.out.yaml [hunk 1]
```diff
@@ -9347,10 +9347,6 @@ spec:
       partition: 0
   serviceName: loki-headless
   revisionHistoryLimit: 10
-  
-  persistentVolumeClaimRetentionPolicy:
```
Comment (arkodg) [inline test/helm/gateway-addons-helm/default.out.yaml:9351]: curious why this is deleted
this is why `make gen-check` is failing

**Comment by zirain** [comment]
@RomainBelorgey can you address arko's comment?

**Comment by RomainBelorgey** [comment]
Sorry got sidetracked to something else :sweat_smile: 
I readded manually `persistentVolumeClaimRetentionPolicy` on the 2 files where it was removed. Hope that it will fix the check

**Comment by arkodg** [comment]
np @RomainBelorgey, I suggest rerunning 'make generate' and commiting those change or looking at the CI to see which test files you've missed and editing them

**Comment by zirain** [comment]
@RomainBelorgey can you fix this, want to land this in `v1.2`.

**Comment by RomainBelorgey** [comment]
> @RomainBelorgey can you fix this, want to land this in `v1.2`.

Normally fixed :crossed_fingers:

**Review by zirain — APPROVED** [review]

**Pull request merged**